### PR TITLE
Detect programming languages used in patches

### PR DESCRIPTION
**Pull request by La0**
I used a simple comparison on filename & extension.
@marco-c & @calixteman , if you know of a Python library doing this, i'll modify my patch.

### CODE REVIEW

**Comment by marco-c** [comment]
LGTM. It might also be useful to have the number of changes per programming language.

I'm not sure why the Travis job failed, I'll retrigger it.

File: libmozdata/utils.py [hunk 1]
```diff
@@ -315,3 +316,32 @@ def get_params_for_url(params):
...
+    """
+    name = os.path.basename(path)
+    extension = os.path.splitext(name)[1][1:]
+    langs = {
+        'Shell': ['sh'],
+        'Xml': ['xml', 'xst'],
+        'Html': ['html', 'xhtml'],
+        'Css': ['css'],
+        'Javascript': ['js', 'jsm'],
+        'Makefile': ['mk', 'Makefile', 'Makefile.am', 'Makefile.in', 'configure.in', 'autoconf.mk.in'],
```
Comment (marco-c) [inline libmozdata/utils.py]: We should also add `moz.build` files, I'm not sure how to categorize them.
Perhaps:
```
'moz.build': ['moz.build']
```

**Comment by La0** [comment]
Thanks for the review @marco-c !
I just rebased the branch and added support for moz.build and ... Python :smirk: .

**Comment by marco-c** [comment]
Great, thanks!

**Comment by calixteman** [comment]
It seems possible to use pygments to guess the programming language:

from pygments.lexers import guess_lexer

with open('/tmp/foo.c', 'rb') as In:
    s = In.read()
    l = guess_lexer(s)
    print l.name

**Comment by marco-c** [comment]
> It seems possible to use pygments to guess the programming language

Interesting. We could use it and then refine the results with our hardcoded list (e.g. pygments probably doesn't detect moz.build files).

**Comment by La0** [comment]
I'll play around with this idea in the next few days...

**Pull request merged**